### PR TITLE
Feature/refactor sparse dot

### PR DESCRIPTION
**Pull request by lessthanoptimal**
Thank you for your contribution to the EJML project.

Before submitting this PR, please make sure:
- [x] Your code is covered by tests
- [x] You ran `./gradlew spotlessJavaApply`

### CODE REVIEW

**Comment by lessthanoptimal** [comment]
@spensonshih this is the refactoring I mentioned. 

There is a significant change that should have been caught in the previous PR. When you took in a DMatrixSparse you effectively re-implemented a conversion to DMatrixSparseCSC. As a result those two functions were removed as being redundant.  Let me know if you agree/disagree with this statement.

**Comment by spensonshih** [comment]
Makes sense @lessthanoptimal

File: main/ejml-dsparse/benchmarks/src/org/ejml/sparse/csc/mult/BenchmarkMatrixVectorMult_DDRM.java [hunk 1]
```diff
@@ -48,35 +36,37 @@
 @Measurement(iterations = 5)
 @State(Scope.Benchmark)
 @Fork(value = 1)
-public class BenchmarkMatrixVectorMult_DDRM_InnerProduct {
+public class BenchmarkMatrixVectorMult_DDRM {
```
Comment (spensonshih) [inline main/ejml-dsparse/benchmarks/src/org/ejml/sparse/csc/mult/BenchmarkMatrixVectorMult_DDRM.java]: Shouldn't this be renamed to `BenchmarkMatrixVectorMult_DSCC`?

Comment (lessthanoptimal) [reply main/ejml-dsparse/benchmarks/src/org/ejml/sparse/csc/mult/BenchmarkMatrixVectorMult_DDRM.java]: correct. thanks for catching that.

**Pull request merged**